x/stakeibc/keeper/lsm.go

```diff
@@ -87,6 +90,15 @@ func (k Keeper) ValidateLSMLiquidStake(ctx sdk.Context, msg types.MsgLSMLiquidSt
 	}, nil
 }
 
+// Generates a unique ID for the LSM token deposit so that, if a slash query is issued,
+// the query callback can be joined back with this tx
+// The key in the store for an LSMTokenDeposit is chainId + denom (meaning, there
```

Comment:
I remember talking about this possibly being an issue way back in early June, this fixes it so that it is still a singleton value behind the key (instead of needing to iterate an array or something) but many can be in progress at once, even from the same staker if they are on different blocks.  So does this also fix the other small issue where a user might try it with a fraction of their IBC'ed LSM tokens before doing the rest but it would have to complete the ICAs before they could do the second part?

Reply:
Right now there can only be one LSM liquid stake for a given denom in progress at once. So the case where a user splits up their tokens and tries to lsm-liquid-stake while they still have an earlier stake in progress will still fail. 

The change above does not modify anything related to this limitation or how we store down the LSM token deposit - it only adds an ID that can be used by the event

Reply:
Oh, I see now.  If we were to apply this same ID logic here to the keeper would that fix this issue by allowing for more than one at a time to be in flight?  Would it be complicated by the way we would want to call it back with the getters since we wouldn't always know the original block height?

Reply:
good question! yeah it would resolve the issue. But imo, it's more a feature than an issue cause if a user has a stuck LSM liquid stake with a denom, I don't think we want to keep allowing LSM liquid stakes with that denom

And yeah that's right, it could potentially get more complicated when we have to use the getters since we don't have the height - although, to be fair, I don't think call the getter directly that often outside of unit tests